include/ck/tensor_operation/gpu/device/impl/device_gemm_multiple_d_xdl_cshuffle_v3.hpp

```diff
@@ -153,6 +158,11 @@ struct DeviceGemmMultiD_Xdl_CShuffle_V3 : public DeviceGemmMultipleDSplitK<ALayo
                 throw std::runtime_error("wrong! GridwiseGemm has invalid setting");
             }
 
+            if((!IsHWSupportAtomicAdd) && (arg.KBatch > 1))
```

Comment:
1. Can we move it to IsSupported?
2. `(CDEShuffleBlockTransferScalarPerVectors{}[I0] > 1) || (sizeof(CDataType) > 2);` Can we improve this condition? Types smaller than 2 are not supported even if ScalarPerVector is divisible by 2 and bf16 is supported only for gfx942